nova/virt/vmwareapi/vmops.py

```diff
@@ -2004,6 +1986,17 @@ def finish_revert_migration(self, context, instance, network_info,
         if vmdk.device:
             self._revert_migration_update_disks(vm_ref, instance, vmdk,
                                                 block_device_info)
+        # Relocate the instance back, if needed
```

Comment:
I think this should be done in (or rather called from) `destroy()`, as that's what the manager calls in `revert_resize()`. There, we would also have the `network_info` we need.

Reply:
_relocate_vm() does not take a destination as parameter, but instead it relocates the VM inside the compute it runs on.
When reverting, we need to move the instance back to the source_compute. Since destroy() is called at dest_compute, this was the convenient place to do this.

Reply:
But `_relocate_vm()` is your function, so you can add that parameter, if you want to. The problem is, that you don't have the migration object and thus cannot easily get the source host to migrate to again. But I still think, that's the place to do it, because afterwards, the migration-status is set to `reverted`, but we didn't do anything, yet. Placement already get the allocations removed and thinks there's free space, while the instance is still taking up some.